server/engine/src/main/java/org/eclipse/lsp/cobol/core/engine/symbols/SymbolAccumulator.java

```diff
@@ -384,7 +410,11 @@ public Optional<SyntaxError> registerParagraphNameNode(
    */
   public CodeBlockReference getCodeBlockReference(ProgramNode programNode, String name) {
     SymbolTable symbolTable = createOrGetSymbolTable(programNode);
-    return symbolTable.getParagraphMap().computeIfAbsent(name, symbolTable.getSectionMap()::get);
+    Map<String, CodeBlockReference> paragraphMap = symbolTable.getParagraphMap();
+    if (paragraphMap.containsKey(name)) {
+      return paragraphMap.get(name);
+    }
+    return symbolTable.getSectionMap().get(name);
```

Comment:
I think it would be better to avoid building those two hashmaps just to ask a question about a key.
I suggest adding `getParagraphByName` and `getSectionByName`.

Note: getting the paragraph by name can have ambiguities, right? The method should (eventually) also have section hint.